cfi/lib/src/cfi_ctr.rs

```diff
@@ -0,0 +1,204 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    cfi_ctr.rs
+
+Abstract:
+
+    File contains CFI Integer and Counter implementations. The counter is based on ideas from
+    Trusted Firmware-M firmware.
+
+References:
+    https://tf-m-user-guide.trustedfirmware.org/design_docs/tfm_physical_attack_mitigation.html
+
+--*/
+
+use crate::cfi::{cfi_panic, CfiPanicInfo};
+use crate::xoshiro::{Xoshiro128, Xoshiro128Reg};
+#[cfg(not(feature = "cfi-test"))]
+use caliptra_common::memory_layout::{CFI_MASK_ORG, CFI_VAL_ORG};
+use core::default::Default;
+
+#[cfg(feature = "cfi-test")]
+static mut CFI_VAL: u32 = 0u32;
+#[cfg(feature = "cfi-test")]
+static mut CFI_MASK: u32 = 0u32;
+
+/// CFI Integer
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+pub struct CfiInt {
+    /// Actual Value
+    val: u32,
+
+    /// Masked Value
+    masked_val: u32,
+}
+
+impl CfiInt {
+    /// Integer mask with high hamming distance
+    const MASK: u32 = 0xA5A5A5A5;
+
+    /// Create integer from raw values
+    fn from_raw(val: u32, masked_val: u32) -> Self {
+        Self { val, masked_val }
+    }
+
+    /// Encode the integer
+    fn encode(val: u32) -> Self {
+        Self {
+            val,
+            masked_val: val ^ Self::MASK,
+        }
+    }
+
+    /// Check if the integer is valid
+    fn is_valid(&self) -> bool {
+        self.val == self.masked_val ^ Self::MASK
+    }
+}
+
+impl Default for CfiInt {
+    /// Returns the "default value" for a type.
+    fn default() -> Self {
+        Self::encode(0)
+    }
+}
+
+/// CFI counter
+pub enum CfiCounter {}
```

Comment:
nitpick: What's the point of making this a type? Shouldn't these just be top level functions in the module?

Reply:
Example use:

```
// nitpick: cfi_counter would be clearer than cfi_ctr.
use cfi::cfi_ctr;

cfi_ctr::reset();
cfi_ctr::corrupt();
```

Reply:
We'll make this change as part of the asym incr/decr proc. macro changes.